### ISSUE TITLE
Don't prevent `Promise.reject()` in the `no-return-wrap` rule

### ISSUE THREAD

**Issue by sindresorhus** [xjamundx/eslint-plugin-promise#50]
Using `return Promise.reject()` is not wrapping, but just another way to reject the promise. I don't think this rule should be concerned about a stylistic issue.

If not, there should at least be an option to turn off requiring `throw`.

---

I personally favor:

```js
myPromise.then(() => Promise.reject(new Error("bad thing")));
```

Over this:

```js
myPromise.then(() => {
	throw new Error('bad thing');
});
```

**Comment by xjamundx** [comment]
I kind of always thought of this as a stylistic rule, but I'm fine making
it configruable!

**Comment by thisgeek** [comment]
I'd be interested in fixing this. Is this how you'd like the option to look?

```javascript
/*eslint promise/no-return-wrap: ["error", { allowReject: true }]*/
myPromise.then(() => {
  return Promise.reject(new Error("bad thing"))
});
```

**Comment by xjamundx** [comment]
Sounds good!
